File: binary_search/0852_peak_index_in_a_mountain_array/0852_peak_index_in_a_mountain_array.go
package main

import "fmt"

func peakIndexInMountainArray(arr []int) int {
	l, r := 0, len(arr)-1
	for l < r {
		mid := l + (r-l+1)/2
		if arr[mid-1] < arr[mid] {
			l = mid
		} else {
			r = mid - 1
		}
	}
	return l
}

func main() {
	arr := []int{0, 1, 0}
	res := peakIndexInMountainArray(arr)
	fmt.Println(res)
}

File: binary_search/0852_peak_index_in_a_mountain_array/0852_peak_index_in_a_mountain_array.py
from typing import List


class Solution:
    def peakIndexInMountainArray(self, A: List[int]) -> int:
        lo, hi = 0, len(A) - 1
        while lo < hi:
            mi = (lo + hi) // 2
            if A[mi] < A[mi + 1]:
                lo = mi + 1
            else:
                hi = mi
        return lo


A = [0,1,0]
res = Solution().peakIndexInMountainArray(A)
print(res)